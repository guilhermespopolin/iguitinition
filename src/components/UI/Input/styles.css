.input {
  outline: none;
  padding: 0.5rem 0.3rem;
  border: none;
  border: 1px solid var(--lightGray);
  border-radius: 3px;
  color: var(--darkGray);
}

.input:focus {
  border: 1px solid var(--primary);
}
