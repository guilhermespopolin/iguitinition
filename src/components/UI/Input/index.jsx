import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import styles from './styles.css'

function Input({ className, ...others }) {
  return <input className={cx(styles.input, className)} {...others} />
}

Input.propTypes = {
  className: PropTypes.string,
}

Input.defaultProps = {
  className: '',
}

export default Input
