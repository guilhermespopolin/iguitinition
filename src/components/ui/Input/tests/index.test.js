import React from 'react'
import { cleanup } from 'react-testing-library'
import renderComponent from '__tests__/utils'

import Input from '..'

afterEach(cleanup)

describe('<Input />', () => {
  it('should render properly', () => {
    const { getByPlaceholderText } = renderComponent(<Input placeholder="Enter some text" />)
    const inputNode = getByPlaceholderText('Enter some text')

    expect(inputNode).toMatchSnapshot()
  })
})

