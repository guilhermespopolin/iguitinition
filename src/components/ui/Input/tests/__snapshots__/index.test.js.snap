// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render properly 1`] = `
.c0 {
  outline: none;
  padding: 0.5rem 0.3rem;
  border: none;
  border: 1px solid #e3e3e3;
  border-radius: 3px;
  color: var(--darkGray);
}

.c0:focus {
  border: 1px solid #4030e8;
}

<input
  class="c0"
  placeholder="Enter some text"
/>
`;
